test/component/login_test.dart

```diff
@@ -1,83 +1,202 @@
+  late FakeFirebaseFirestore fakeFireStore;
+  late CollectionReference<Map<String, dynamic>> userCollection;
+  late UserRepositoryService userRepo;
 
-  final mockedProxima = ProviderScope(
-    overrides: firebaseAuthMocksOverrides,
-    child: const ProximaApp(),
-  );
+  setupFirebaseAuthMocks();
 
   setUpAll(() async {
     await Firebase.initializeApp();
   });
 
-  testWidgets("Login to Create Account Page to Home Page flow", (tester) async {
-    await tester.pumpWidget(mockedProxima);
-    await tester.pumpAndSettle();
-
-    // Check for the logo on the Login Page
-    final logoFinder = find.byKey(LoginPage.logoKey);
-    expect(logoFinder, findsOneWidget);
-
-    // Check for the slogan on the Login Page
-    final sloganFinder = find.text(LoginPage.tagLineText);
-    expect(sloganFinder, findsOneWidget);
-
-    final loginButton = find.byKey(LoginButton.loginButtonKey);
-    // Check that the login button is displayed and contains the "Login" text
-    expect(
-      find.descendant(
-        of: loginButton,
-        matching: find.text("Sign in with Google"),
-      ),
-      findsOneWidget,
-    );
-
-    await tester.tap(loginButton);
-    await tester.pumpAndSettle();
-
-    // Check that pressing login redirects to the create account page
-    final createAccountPage = find.byType(CreateAccountPage);
-    expect(createAccountPage, findsOneWidget);
-
-    // And that pushing the confirm button redirects to the home page
-    final confirmButton = find.byKey(CreateAccountPage.confirmButtonKey);
-    await tester.tap(confirmButton);
-    await tester.pumpAndSettle();
-
-    // We must now be on the home page
-    final homePage = find.byType(HomePage);
-    expect(homePage, findsOneWidget);
+  group("Existing user data in repository testing", () {
+    final expectedUser = testingUserFirestore;
+
+    setUp(() async {
+      fakeFireStore = FakeFirebaseFirestore();
+      userCollection = fakeFireStore.collection(UserFirestore.collectionName);
+      userRepo = UserRepositoryService(
+        firestore: fakeFireStore,
+      );
+      await userCollection
+          .doc(expectedUser.uid.value)
+          .set(expectedUser.data.toDbData());
+    });
+
+    ProviderScope getMockedProxima() {
+      return ProviderScope(
+        overrides: [
+          ...firebaseAuthMocksOverrides,
+          userRepositoryProvider.overrideWithValue(userRepo),
+        ],
+        child: const ProximaApp(),
+      );
+    }
+
+    testWidgets("Login and get to home page flow", (tester) async {
+      await tester.pumpWidget(getMockedProxima());
+      await tester.pumpAndSettle();
+
+      final loginButton = find.byKey(LoginButton.loginButtonKey);
+      await tester.tap(loginButton);
+      //Needs a delay to allow the existance check to complete
+      await tester.pumpAndSettle(delayNeededForAsyncFunctionExecution);
+
+      final loginButtonStillHere = find.byType(HomePage);
```

Comment:
By humble curiosity, what was the reasoning behind this variable name (same question for all the `loginButtonStillHere` variables in this test file)?

Reply:
Good catch, I think it was for me during my test creation to debug and I forgot to rename it.